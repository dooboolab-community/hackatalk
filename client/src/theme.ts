import type {DoobooTheme} from 'dooboo-ui';

/* eslint-disable */
export const colors = {
  brand: '#1E6EFA',
  paleBlue05: '#E2F2FF',
  paleBlue10: '#8BC9FF',
  green10: '#A1EEDB',
  green30: '#00D4AB',
  green50: '#28DB98',
  gray05: '#F3F9FF',
  success: '33ff2f',
  danger: '#ff002e',
  warning: '#f2df0f',
  info: '#3a74e7',
  light: '#ededed',
  dark: '#151C22',
  messagePrimary: '#8BC9FF',
  messageSecondary: '#E2F2FF',
  apple: '#151E22',
  google: '#E04238',
  facebook: '#345997',
};
/* eslint-enable */

export const light = {
  background: 'white',
  paper: '#F2F5F6',
  disabled: '#C4C4C4',
  line: '#gray05',
  text: '#000',
  textLight: '#1B1B1B',
  textContrast: 'white',
  placeholder: '#6D6D6D',
  header: colors.brand,
  card: 'white',
  active: colors.brand,
  primary: colors.brand,
  primaryLight: colors.paleBlue10,
  secondary: colors.green10,
  button: colors.brand,
};

export type Theme = typeof light & DoobooTheme;

export const dark = {
  background: '#000',
  paper: colors.dark,
  disabled: '#515151',
  line: '#000',
  text: 'white',
  textLight: '#D3D3D3',
  textContrast: '#000',
  placeholder: '#6D6D6D',
  header: '#000',
  card: colors.dark,
  active: colors.paleBlue10,
  primary: '#fff',
  primaryLight: '#151A25',
  secondary: colors.green30,
  button: colors.paleBlue10,
};
