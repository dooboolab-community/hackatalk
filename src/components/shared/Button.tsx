import {
  ActivityIndicator,
  ImageSourcePropType,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';

import React from 'react';
import styled from 'styled-components/native';

interface StyledElement {
  white?: boolean;
}

interface ButtonContainer {
  width: number;
  height: number;
}

interface Props {
  testID: string;
  containerStyle?: ViewStyle;
  isWhite: boolean;
  isLoading: boolean;
  isDisabled: boolean;
  onPress: () => void;
  imgLeftSrc?: ImageSourcePropType;
  indicatorColor: string;
  activeOpacity: number;
  children: string;
  width: number;
  height: number;
}

const StyledButtonContainer = styled.View<ButtonContainer>`
  width: ${({ width }): string => `${width}px`};
  height: ${({ height }): string => `${height}px`};
  justify-content: center;
  align-items: center;
`;

const StyledButton = styled.View<StyledElement>`
  background-color: ${({ white, theme }): string =>
    white ? theme.btnPrimaryLight : theme.btnPrimary};
  border-color: ${({ white, theme }): string =>
    white ? theme.btnPrimary : theme.btnPrimaryLight};
  border-radius: 4px;
  border-width: 1px;
  align-items: center;
  justify-content: center;
`;

const StyledButtonDisabled = styled.View`
  background-color: rgb(243, 243, 243);
  align-self: center;
  border-radius: 4px;
  border-width: 2px;
  border-color: #333;
  align-items: center;
  justify-content: center;
`;

const StyledText = styled.Text<StyledElement>`
  font-size: 14px;
  font-weight: bold;
  color: ${({ white, theme }): string =>
    white ? theme.btnPrimaryLightFont : theme.btnPrimaryFont};
`;

const StyledImageLeft = styled.Image`
  width: 24px;
  height: 24px;
  position: absolute;
  left: 16px;
`;

function Button(props: Props): React.ReactElement {
  const {
    testID,
    isDisabled,
    isWhite,
    activeOpacity,
    onPress,
    isLoading,
    width,
    height,
    indicatorColor,
    imgLeftSrc,
    children,
  } = props;

  return (
    <>
      {isDisabled ? (
        <StyledButtonDisabled testID={testID}>
          <StyledText>{children}</StyledText>
        </StyledButtonDisabled>
      ) : (
        <TouchableOpacity
          testID={testID}
          activeOpacity={activeOpacity}
          onPress={onPress}
        >
          <StyledButton white={isWhite}>
            <StyledButtonContainer width={width} height={height}>
              {isLoading ? (
                <ActivityIndicator size="small" color={indicatorColor} />
              ) : imgLeftSrc ? (
                <StyledImageLeft source={imgLeftSrc} />
              ) : (
                undefined
              )}
              {!isLoading ? (
                <StyledText white={isWhite}>{children}</StyledText>
              ) : (
                undefined
              )}
            </StyledButtonContainer>
          </StyledButton>
        </TouchableOpacity>
      )}
    </>
  );
}

Button.defaultProps = {
  isWhite: false,
  isLoading: false,
  isDisabled: false,
  indicatorColor: 'white',
  activeOpacity: 0.5,
  width: 136,
  height: 60,
};

export default Button;
