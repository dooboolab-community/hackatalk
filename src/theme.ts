/* eslint-disable */
export const colors = {
  light: '#ffffff',
  dark: '#151A25',
  backgroundDark: '#151E22',
  deepDark: '#000000',
  blue_80: '#2F38C7',
  blue_70: '#1E6EFA',
  blue_60: '#217DFE',
  paleBlue_10: '#8bc9ff',
  paleBlue_05: '#e2f2ff',
  green_30: '#00D4AB',
  green_10: '#A1EEDB',
  gray_80: '#676c7a',
  gray_70: '#545966',
  gray_60: '#676c7a',
  gray_50: '#8f94a3',
  gray_40: '#AFB4C3',
  gray_30: '#d3d8e8',
  gray_05: '#f3f9ff',
  apple: '#151E22',
  google: '#E04238',
  facebook: '#345997',
};
/* eslint-enable */

export const light = {
  background: colors.light,
  backgroundDark: colors.paleBlue_05,
  itemBackground: colors.light,
  btnPrimary: colors.blue_70,
  btnPrimaryFont: colors.light,
  btnPrimaryLight: colors.light,
  btnPrimaryLightFont: colors.blue_70,
  textDisabled: colors.gray_50,
  btnDisabled: colors.gray_30,
  fontColor: colors.dark,
  fontSubColor: colors.blue_70,
  labelColor: colors.dark,
  tintColor: colors.green_30,
  lineColor: colors.gray_30,
  indicatorColor: colors.blue_70,
  inactiveColor: colors.gray_60,
  primary: colors.blue_70,
  primaryLight: colors.paleBlue_10,
  headerFont: colors.light,
  searchBackground: colors.gray_05,
  messageBackground: colors.light,
  status: colors.green_10,
  placeholder: colors.gray_60,
  focused: colors.blue_70,
  placeholderFocused: colors.blue_70,
  socialBtn: colors.paleBlue_05,
  link: colors.blue_70,
  myMessageBackground: colors.paleBlue_10,
  myMessageText: colors.dark,
  peerMessageBackground: colors.paleBlue_05,
  peerMessageText: colors.dark,
  modalBtnBackground: colors.light,
  modalBtnPrimaryFont: colors.blue_80,
  modalBtnFont: colors.gray_60,
  appleIcon: colors.apple,
  appleText: colors.apple,
  appleBackground: colors.light,
  facebookIcon: colors.light,
  facebookText: colors.light,
  facebookBackground: colors.facebook,
  googleIcon: colors.light,
  googleText: colors.light,
  googleBackground: colors.google,
  errorTitle: colors.gray_60,
  errorBody: colors.gray_50,
  fab: colors.blue_70,
};
export type Theme = typeof light;
export const dark = {
  background: colors.deepDark,
  backgroundDark: colors.deepDark,
  itemBackground: colors.dark,
  btnPrimary: colors.paleBlue_10,
  btnPrimaryFont: colors.deepDark,
  btnPrimaryLight: colors.dark,
  btnPrimaryLightFont: colors.paleBlue_10,
  textDisabled: colors.gray_80,
  btnDisabled: colors.gray_40,
  fontColor: colors.light,
  fontSubColor: colors.paleBlue_10,
  labelColor: colors.light,
  tintColor: colors.green_10,
  lineColor: colors.light,
  indicatorColor: colors.paleBlue_10,
  inactiveColor: colors.gray_50,
  primary: colors.deepDark,
  primaryLight: colors.dark,
  headerFont: colors.light,
  searchBackground: colors.gray_80,
  messageBackground: colors.dark,
  status: colors.green_30,
  placeholder: colors.gray_50,
  focused: '#ffffff',
  placeholderFocused: colors.paleBlue_10,
  socialBtn: colors.gray_05,
  link: colors.paleBlue_05,
  myMessageBackground: colors.deepDark,
  myMessageText: colors.light,
  peerMessageBackground: colors.paleBlue_05,
  peerMessageText: colors.dark,
  modalBtnBackground: colors.gray_80,
  modalBtnPrimaryFont: colors.paleBlue_10,
  modalBtnFont: colors.gray_40,
  appleIcon: colors.apple,
  appleText: colors.apple,
  appleBackground: colors.light,
  facebookIcon: colors.facebook,
  facebookText: colors.dark,
  facebookBackground: colors.light,
  googleIcon: colors.google,
  googleText: colors.dark,
  googleBackground: colors.light,
  errorTitle: colors.paleBlue_10,
  errorBody: colors.gray_40,
  fab: colors.paleBlue_10,
};
