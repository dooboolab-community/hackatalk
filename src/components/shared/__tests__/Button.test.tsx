import 'react-native';

import * as React from 'react';

import {
  RenderResult,
  act,
  fireEvent,
  render,
} from '@testing-library/react-native';
import { ThemeType, createTheme } from '../../../theme';

import Button from '../Button';
import { ThemeProvider } from 'styled-components/native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

let props: any;
let component: React.ReactElement;
let testingLib: RenderResult;

describe('[Button]', () => {
  let cnt = 1;

  beforeEach(() => {
    props = {
      onClick: () => cnt++,
      testID: 'btn',
    };
    component = (
      <ThemeProvider theme={createTheme(ThemeType.LIGHT)}>
        <Button {...props} />
      </ThemeProvider>
    );
  });

  it('renders without crashing', () => {
    const rendered = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  describe('interactions', () => {
    beforeEach(() => {
      testingLib = render(component);
    });

    it('should simulate onClick', () => {
      const btn = testingLib.queryByTestId('btn');
      act(() => {
        fireEvent.press(btn);
        fireEvent.press(btn);
      });
      expect(cnt).toBe(3);
    });
  });
});
