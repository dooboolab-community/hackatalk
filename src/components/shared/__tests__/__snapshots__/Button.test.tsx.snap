// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Button] renders without crashing 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="btn"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(58,139,255)",
          "borderColor": "white",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
        },
      ]
    }
    white={false}
  >
    <View
      height={60}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 60,
            "justifyContent": "center",
            "width": 136,
          },
        ]
      }
      width={136}
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
            },
          ]
        }
        white={false}
      />
    </View>
  </View>
</View>
`;
